projects/storefrontlib/shared/components/item-counter/item-counter.service.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ElementRef, Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ItemCounterService {
+  protected counter = 1;
+
+  protected isInitialProductCounter(
+    input?: ElementRef<HTMLInputElement>
+  ): boolean {
+    const wrapperClass = 'cx-counter-stock';
```

Comment:
[QUESTION]

I am having a hard time understanding this custom logic, specifically checking if an ancestor element has a class `cx-counter-stock` and how it relates to the counter being an "initialProduct". Could you help me understand?
I guess it's a workaround for some nasty issue. If yes, what other alternative approaches did we consider and what were their pros and cons?

Reply:
[OUTDATED] We dropped such service.